Don't need to downgrade optimisation on ppc/ia64 with gcc 4.x

diff -Naur --exclude=debian perl-5.10.0.orig/hints/linux.sh perl-5.10.0/hints/linux.sh
--- perl-5.10.0.orig/hints/linux.sh	2007-11-18 02:26:24.000000000 +1100
+++ perl-5.10.0/hints/linux.sh	2007-11-26 02:57:15.000000000 +1100
@@ -113,23 +113,6 @@
 # use -O2 by default ; -O3 doesn't seem to bring significant benefits with gcc
 '')
     optimize='-O2'
-    case "`uname -m`" in
-        ppc*)
-            # on ppc, it seems that gcc (at least gcc 3.3.2) isn't happy
-            # with -O2 ; so downgrade to -O1.
-            optimize='-O1'
-        ;;
-        ia64*)
-            # This architecture has had various problems with gcc's
-            # in the 3.2, 3.3, and 3.4 releases when optimized to -O2.  See
-            # RT #37156 for a discussion of the problem.
-            case "`${cc:-gcc} -v 2>&1`" in
-            *"version 3.2"*|*"version 3.3"*|*"version 3.4"*)
-                ccflags="-fno-delete-null-pointer-checks $ccflags"
-            ;;
-            esac
-        ;;
-    esac
     ;;
 esac
 
